source/slang-capture-replay/replay/parameter-decoder.cpp

```diff
@@ -0,0 +1,215 @@
+#include <string.h>
+#include "parameter-decoder.h"
+
+namespace SlangCapture
+{
+    size_t ParameterDecoder::decodeString(const uint8_t* buffer, int64_t bufferSize, PointerDecoder<char*>& typeDecoder)
+    {
+        SLANG_CAPTURE_ASSERT((buffer != nullptr) && (bufferSize > 0));
+
+        if (bufferSize < (int64_t)sizeof(uint32_t))
+        {
+            return 0;
+        }
+
+        uint32_t stringLength = 0;
+        size_t readByte = 0;
+        readByte += decodeUint32(buffer, bufferSize - readByte, stringLength);
+
+        SLANG_CAPTURE_ASSERT(bufferSize >= (int64_t)(readByte + stringLength));
+
+        if (stringLength == 0)
+        {
+            return readByte;
+        }
+
+        uint8_t* data = (uint8_t*)typeDecoder.allocate(stringLength + 1);
+        memcpy(data, buffer + readByte, stringLength);
```

Comment:
The allocation can fail and return null pointer.
The behavior of memcpy will be undefined when data is nullptr.